### PR TITLE
fix: removed unnecessary stripe call from boot sequence

### LINKED ISSUE #17510: Stripe API call is breaking all local setups
#### Describe your problem and - if possible - how to reproduce it

When following the instructions in the contributors guide there seems to be an unaccounted for issue.
With a fresh build, if you attempt to execute `npm run develop` you'll be presented with an error.
To resolve the error you have to add a valid Strip public and private key to the .env file before executing `npm run build`. I'm not sure if the CONTRIBUTING.md should just be updated or if freeCodeCamp/server/boot/donate.js could be updated to prevent this issue.

Here is the error:

```
/home/charles/development/freeCodeCamp/server/boot/donate.js:62 throw err; ^ Error: Invalid API Key provided: sk_from_*****_*****oard at Constructor._Error (/home/charles/development/freeCodeCamp/node_modules/stripe/lib/Error.js:12:17) at Constructor (/home/charles/development/freeCodeCamp/node_modules/stripe/lib/utils.js:133:13) at new Constructor (/home/charles/development/freeCodeCamp/node_modules/stripe/lib/utils.js:133:13) at IncomingMessage.<anonymous> (/home/charles/development/freeCodeCamp/node_modules/stripe/lib/StripeResource.js:164:21) at emitNone (events.js:111:20) at IncomingMessage.emit (events.js:208:7) at endReadableNT (_stream_readable.js:1056:12) at _combinedTickCallback (internal/process/next_tick.js:138:11) at process._tickDomainCallback (internal/process/next_tick.js:218:9) [08:19:31] [nodemon] app crashed - waiting for file changes before starting...
```

### PR DESCRIPTION
**Pull request by cmoss3**
#### Pre-Submission Checklist

- [x] Your pull request targets the `staging` branch of freeCodeCamp.
- [x] Branch starts with either `fix/`, `feature/`, or `translate/` (e.g. `fix/signin-issue`)
- [x] You have only one commit (if not, [squash](http://forum.freecodecamp.org/t/how-to-squash-multiple-commits-into-one-with-git/13231) them into one commit).
- [x] All new and existing tests pass the command `npm test`. Use `git commit --amend` to amend any fixes.

#### Type of Change

- [x] Small bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Add new translation (feature adding new translations)

#### Checklist:

- [ ] Tested changes locally.
- [x] Addressed currently open issue (replace XXXXX with an issue no in next line)

Closes #17510

### CODE REVIEW

**Review by Bouncey — CHANGES_REQUESTED** [review]
Thanks @cmoss3 for your work on this. I think instead of leaving the setup of stripe to the first request to each server instance (we run more than one in production), we should still do the setup in the boot phase of the app. 

We can still use your `connectToStripe` function, but we should have it return a promise. Let me explain what I have in my head

```js

export default function donateBoot(app, done) {
  if (!stripeAPIKey) {
    if (NODE_ENV === 'production') {
      throw new Error('Stripe API keys required!');
      // if we are in production and have no keys, throw
    }
    console.info('No Stripe API keys found, this can be safely ignored in development';);
    return done();
    // if we don't have an API Key, we don't setup the donate routes, this is OK for development
  }
  // ... setup route handlers and routers, then...
  connectToStripe().then(done);
}
```

I hope this makes sense.

**Comment by cmoss3** [comment]
Hey @Bouncey, I updated the PR with the changes you mentioned.  I did make the connectToStripe function return a promise but it's setup so that the "done" function will be called before there is a response from the Stripe API, but after the requests were made. I hope that's what you expected, if not just let me know.

Also, still haven't been able to test the /external/donate/charge-stripe API. I did try testing it with Postman but I was a bit confused on what parameters should be passed and also I think I need to be logged in to get a valid JWT.

File: server/boot/donate.js [hunk 1]
```diff
@@ -1,121 +1,138 @@
 import Stripe from 'stripe';
-
 import keys from '../../config/secrets';
 
-const stripe = Stripe(keys.stripe.secret);
+export default function(app, done) {
```
Comment (Bouncey) [inline server/boot/donate.js]: `export default function donateBoot(app, done) {`

File: server/boot/donate.js [hunk 2]
```diff
@@ -1,121 +1,138 @@
...
+            .catch(err => {
+                if (err.type === 'StripeCardError') {
+                    return res.status(402).send({ error: err.message });
+                }
+                return res.status(500).send({ error: 'Donation Failed' });
+            });
+    }
 
-  api.post('/charge-stripe', createStripeDonation);
+    const secretInvalid = keys.stripe.secret === 'sk_from_stipe_dashboard';
```
Comment (Bouncey) [inline server/boot/donate.js]: We should check to see if we have keys as well. 

```js
const secretInvalid = !keys.stripe.secret || keys.stripe.secret === 'sk_from_stipe_dashboard';
const publicInvalid = !keys.stripe.public || keys.stripe.public === 'pk_from_stipe_dashboard';
```

**Review by Bouncey — CHANGES_REQUESTED** [review]
Thanks for working on this @cmoss3, you have introduced some extra whitespace in this file.

VS Code does this to me sometimes also. please convert them back to spaces (2).

**Comment by cmoss3** [comment]
Hrm, seems like the build is running into some linting error. 
Locally I ran `npm run test`, `npm run lint` and `npm run lint-js` none of which produced any errors.
Any suggestions?

**Comment by cmoss3** [comment]
Yikes, had to update one more time because I forgot to name the export function. O_O'

@Bouncey These changes are done though. Looks like there's still some issues with the build but all tests are passing locally.

**Comment by Nirajn2311** [comment]
@cmoss3 the linking error will be fixed once #17609 is merged. It is caused due to wrong location of eslint when the command gets executed.

**Review by Bouncey — APPROVED** [review]
Awesome! Thanks @cmoss3

**Pull request merged**